internal/app/shortener/delivery/rest/rest.go

```diff
@@ -0,0 +1,65 @@
+package rest
+
+import (
+	"context"
+	"github.com/gin-gonic/gin"
+	"github.com/go-chi/chi/v5"
+	"github.com/rltsv/urlcutter/internal/app/shortener/repository"
+	"github.com/rltsv/urlcutter/internal/app/shortener/usecase/shortener"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+type HandlerShortener struct {
+	useCase shortener.Usecase
+	*chi.Mux
+}
+
+func NewHandlerShortener(shortenerUseCase shortener.Usecase) *HandlerShortener {
+	return &HandlerShortener{
+		useCase: shortenerUseCase,
+		Mux:     chi.NewMux(),
+	}
+}
+
+func (hs *HandlerShortener) CreateShortLink(c *gin.Context) {
+
+	ctx := context.Background()
```

Comment:
Здесь и везде:

```suggestion
	ctx := c.Request.Context()
```
https://pkg.go.dev/net/http#Request.Context

Reply:
Принял